In [21]:
from nltk.translate.bleu_score import sentence_bleu
import Levenshtein
import gensim.downloader as api

Leemos ambos resúmenes

In [15]:
with open("summary-1-flan-ul2--article1.txt", "r", encoding="utf-8") as file:
    reference_summary = file.read().lower().split(' ')
print(reference_summary)
print(len(reference_summary))

['people', 'are', 'using', 'ai', 'chatbots', 'to', 'fill', 'junk', 'websites', 'with', 'ai-generated', 'text', 'that', 'attracts', 'paying', 'advertisers,', 'according', 'to', 'a', 'new', 'report', 'from', 'the', 'media', 'research', 'organization', 'newsguard', 'that', 'was', 'shared', 'exclusively', 'with', 'mit', 'technology', 'review.', 'over', '140', 'major', 'brands', 'are', 'paying', 'for', 'ads', 'that', 'end', 'up', 'on', 'unreliable', 'ai-written', 'sites,', 'likely', 'without', 'their', 'knowledge.', 'ninety', 'percent', 'of', 'the', 'ads', 'from', 'major', 'brands', 'found', 'on', 'these', 'ai-generated', 'news', 'sites', 'were', 'served', 'by', 'google,', 'though', 'the', 'company’s', 'own', 'policies', 'prohibit', 'sites', 'from', 'placing', 'google-served', 'ads', 'on', 'pages', 'that', 'include', '“spammy', 'automatically', 'generated', 'content.”', 'the', 'practice', 'threatens', 'to', 'hasten', 'the', 'arrival', 'of', 'a', 'glitchy,', 'spammy', 'internet', 'that', 'is

In [16]:
with open("summary-2-flan-ul2--article1.txt", "r", encoding="utf-8") as file:
    candidate_summary = file.read().lower().split(' ')
print(candidate_summary)
print(len(candidate_summary))

['a', 'new', 'report', 'finds', 'that', 'sites', 'run', 'with', 'ai-generated', 'content', 'serve', 'ads', 'from', 'major', 'brands,', 'which', 'mostly', 'come', 'from', 'google.', 'some', 'of', 'those', 'sites', 'contained', 'dangerous', 'misinformation.', 'and', 'this', 'is', 'just', 'getting', 'started.', 'more', 'could', 'be', 'on', 'the', 'way.', '"the', 'opaque', 'nature', 'of', 'programmatic', 'advertising', 'has', 'inadvertently', 'turned', 'major', 'brands', 'into', 'unwitting', 'supporters."\n']
53


In [18]:
# BLEU for syntactic similarity
bleu_score = sentence_bleu(reference_summary, candidate_summary)
print(bleu_score)

6.752107625974243e-232


No parece ser muy útil el BLEU score ya que no encuentra conjuntos de palabras consecutivas que sean iguales, por lo que el score sale casi 0

In [19]:
# Levenshtein distance
lev_dist = Levenshtein.distance(reference_summary, candidate_summary)
print(f"Levenshtein Distance: {lev_dist}")


Levenshtein Distance: 107


La distancia de levenshtein indica que habría que hacer 107 cambios para pasar de un resumen a otro, lo cual indica que no se parecen demasiado al ser resumenes de 118 y 53 palabras

In [23]:
# Use pretrained word embeddings
word_vectors = api.load("word2vec-google-news-300")  # Large model (~1.5GB)

# Compute WMD distance to measure distance based on embeddings
distance = word_vectors.wmdistance(reference_summary, candidate_summary)
print(f"WMD Distance: {distance}")



WMD Distance: 0.840365499455094


Una distancia de 0.84 indica un nivel relativamente alto de disparidad entre los documentos, incluso haciendo uso de word embeddings, por lo que se puede concluir que los documentos no son muy similares